fdbserver/MoveKeys.actor.cpp

```diff
@@ -1210,22 +1211,27 @@ void seedShardServers(Arena& arena, CommitTransactionRef& tr, vector<StorageServ
 	tr.read_snapshot = 0;
 	tr.read_conflict_ranges.push_back_deep(arena, allKeys);
 
-	for (int s = 0; s < servers.size(); s++) {
-		tr.set(arena, serverTagKeyFor(servers[s].id()), serverTagValue(server_tag[servers[s].id()]));
-		tr.set(arena, serverListKeyFor(servers[s].id()), serverListValue(servers[s]));
+	for (auto& s : servers) {
+		tr.set(arena, serverTagKeyFor(s.id()), serverTagValue(server_tag[s.id()]));
+		tr.set(arena, serverListKeyFor(s.id()), serverListValue(s));
 	}
 
 	std::vector<Tag> serverTags;
-	for (int i = 0; i < servers.size(); i++)
-		serverTags.push_back(server_tag[servers[i].id()]);
+	std::vector<UID> serverSrcUID;
+	for (auto& s : servers) {
+		serverTags.push_back(server_tag[s.id()]);
+		serverSrcUID.push_back(s.id());
+	}
+
+	auto ksValue = CLIENT_KNOBS->TAG_ENCODE_KEY_SERVERS ? keyServersValue(serverTags)
```

Comment:
Suppose a 6.3.11 cluster has TAG_ENCODE_KEY_SERVERS set.
DBA set TAG_ENCODE_KEY_SERVERS to false (say by accident since setting knob seems a not-risky operation), won't data be corrupted. 

Guarding against such knob-change seems hard. Maybe we should remove the knob and use version to guard agains the encoding format?

Reply:
@sfc-gh-etschannen thoughts on above?